src/domain/Contact.js

```diff
@@ -381,8 +381,12 @@ export default class Contact {
       emails.push(...formattedEmails);
     }
 
-    if (single.numbers) {
-      const formattedPhones = single.numbers.map(phone => ({ label: 'mobile', number: phone }));
+    if (single.numbers_by_label) {
+      const formattedPhones = Object.keys(single.numbers_by_label).map(label =>
+        ({ label, number: single.numbers_by_label[label] }));
+      numbers.push(...formattedPhones);
+    } else if (single.numbers) {
+      const formattedPhones = single.numbers.map(phone => ({ number: phone }));
       numbers.push(...formattedPhones);
```

Comment:
`numbers.concat(formattedPhones)` ?

Reply:
:man_facepalming:

Reply:
Whoa, i take that :man_facepalming: back -- i think i still like that `push` better. What about instead of this:
```
const formattedPhones = single.numbers.map(phone => ({ number: phone }));
numbers.push(...formattedPhones);
```
(which is admittedly a little yucky as the :man_facepalming: suggests), we do this instead:
```
single.numbers.map(phone => numbers.push({ number: phone }));
```
Not to fond of `concat` as it will impose either a `let` on the array, or adding a few transitory arrays throughout the code.

Reply:
Ok but with a forEach instead of a map (map suggest that you a returning someting.
```js
single.numbers.forEach(phone => numbers.push({ number: phone }));

```